hpx/runtime/parcelset/parcel.hpp

```diff
@@ -149,182 +104,109 @@ namespace hpx { namespace parcelset
-            naming::id_type const& dests,
-            naming::address const& addrs,
-            Action,
-            Args &&... args
-        )
-#if defined(HPX_SUPPORT_MULTIPLE_PARCEL_DESTINATIONS)
-          : dests_(1, dests),
-            addrs_(1, addrs),
-#else
-          : dests_(dests),
-            addrs_(addrs),
-#endif
-            action_(new actions::transfer_action<Action>(std::forward<Args>(args)...))
-        {
-            HPX_ASSERT(is_valid());
-        }
-
-#if defined(HPX_SUPPORT_MULTIPLE_PARCEL_DESTINATIONS)
-        template <
-            typename Action
-          , typename...Args
-          , typename Enable =
-                typename std::enable_if<traits::is_action<Action>::value>::type
-          >
-        parcel(
-            std::vector<naming::id_type> const& dests,
-            std::vector<naming::address> const& addrs,
-            Action,
-            Args &&... args
-        )
-          : dests_(dests),
-            addrs_(addrs),
-            action_(new actions::transfer_action<Action>(std::forward<Args>(args)...))
-        {
-            HPX_ASSERT(is_valid());
-        }
-#endif
-
-        template <
-            typename Continuation
-          , typename Action
-          , typename...Args
-          , typename Enable
-                = typename std::enable_if<
-                    traits::is_action<Action>::value
-                 && traits::is_continuation<Continuation>::value
-                 && !std::is_same<
-                        Continuation
-                      , std::unique_ptr<actions::continuation>
-                    >::value
-                >::type
-          >
-        parcel(
-            naming::id_type const& dests,
-            naming::address const& addrs,
-            Continuation && cont,
-            Action,
-            Args &&... args
-        )
-#if defined(HPX_SUPPORT_MULTIPLE_PARCEL_DESTINATIONS)
-          : dests_(1, dests),
-            addrs_(1, addrs),
-#else
-          : dests_(dests),
-            addrs_(addrs),
-#endif
-            cont_(
-                new typename util::decay<Continuation>
-                    ::type(std::forward<Continuation>(cont))
-            ),
-            action_(new actions::transfer_action<Action>(std::forward<Args>(args)...))
-        {
-            HPX_ASSERT(is_valid());
-        }
-
-        template <
-            typename Action
-          , typename...Args
-          , typename Enable
-                = typename std::enable_if<
-                    traits::is_action<Action>::value
-                >::type
-          >
+    private:
         parcel(
-            naming::id_type const& dests,
-            naming::address const& addrs,
+            naming::gid_type&& dest,
+            naming::address&& addr,
             std::unique_ptr<actions::continuation> cont,
-            Action,
-            Args &&... args
+            std::unique_ptr<actions::base_action> act,
+            bool action_args_bitwise = false
```

Comment:
I removed it completely again. It has been shown that it didn't have any noticable effect at all.